caller.Lookup panics in forked repository

**Describe the problem**

Build or run tests in forked repository causes a panic.

**To Reproduce**

```shell
make testlogic FILES="alias_types"
```

```plain
panic: cannot find crdb path ("github.com/cockroachdb/cockroach/pkg/util/caller") inside file path ("/Users/aofedorov/go/src/github.com/alxfv/cockroach/pkg/util/caller")

goroutine 1 [running]:
github.com/cockroachdb/cockroach/pkg/util/caller.findFileAndPackageRoot(0x7c8c132, 0x2, 0xc0000000d4, 0xc0002b7c20, 0x0)
	/Users/aofedorov/go/src/github.com/alxfv/cockroach/pkg/util/caller/resolver.go:117 +0x758
github.com/cockroachdb/cockroach/pkg/util/caller.glob..func1(0x7c8c132)
	/Users/aofedorov/go/src/github.com/alxfv/cockroach/pkg/util/caller/resolver.go:167 +0x26
github.com/cockroachdb/cockroach/pkg/util/caller.init()
	/Users/aofedorov/go/src/github.com/alxfv/cockroach/pkg/util/caller/resolver.go:174 +0x58
```

**Environment:**
```
Build Tag:        v21.2.0-alpha.00000000-1669-g19865dd4d2-dirty
Build Time:
Distribution:     CCL
Platform:         darwin amd64 (x86_64-apple-darwin20.2.0)
Go Version:       go1.16.4
C Compiler:       Apple LLVM 12.0.5 (clang-1205.0.22.9)
Build Commit ID:  19865dd4d29e87212f36b237aaaf543a7d9d55f5
Build Type:       development
```

--- a/pkg/util/caller/resolver.go
+++ b/pkg/util/caller/resolver.go
@@ -85,6 +85,14 @@ func findFileAndPackageRoot() (ok bool, crdbPath string, srcRoot string) {
 	// root is its immediate parent directory.
 	root := path.Dir(frame.File)
 
+	// if it is a forked repository, we need to normalize root
+	// For example:
+	//   /home/kena/src/go/src/github.com/kena/cockroach/pkg/util/caller/resolver.go
+	// should be
+	//   /home/kena/src/go/src/github.com/cockroachdb/cockroach/pkg/util/caller/resolver.gos
+	exp := regexp.MustCompile("github.com/([^/]+)/cockroach")
+	root = exp.ReplaceAllString(root, "github.com/cockroachdb/cockroach")
+
 	// Coverage tests report back as `[...]/util/caller/_test/_obj_test`;
 	// strip back to this package's directory.
 	if !strings.HasSuffix(root, "/caller") {

